pkgs/development/python-modules/django-rapyd-modernauth/default.nix

```
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, asgiref
+, django
+, python-dotenv
+, pytz
+, setuptools
+, sqlparse
+, zipp
+}:
+
+buildPythonPackage rec {
+  pname = "django-rapyd-modernauth";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "karthicraghupathi";
+    repo = "django_rapyd_modernauth";
+    rev = "v${version}";
+    hash = "sha256-9IIW6Xsaj2+/SR2CIX4fnUHlAehmLulpBAC7YB3ZrZE=";
+  };
+
+  # Tests are not part of the released package, so we cannot run them
+  doCheck = true;
+  checkPhase = ''
+    echo 'LOG_LEVEL="INFO"' > .env
+    echo DJANGO_SECRET_KEY="$(base64 /dev/urandom | head -c50)" >> .env
+    python3 ./testrunner.py
```

Could we avoid this by using pytest-django?

---

I don't know that we can. The upstream package does not have pytests. It just has some manual commands that get run in the `testrunner.py` main function. It's quite non-standard in its test setup.

---

```suggestion
    echo 'LOG_LEVEL="INFO"' > .env
    echo DJANGO_SECRET_KEY="$(base64 /dev/urandom | head -c50)" >> .env
    ${python.interpreter} ./testrunner.py
```
then we want to do this